### PR TITLE
Scheduler first fit

### LINKED ISSUE #122057: Scheduler evaluates many nodes even if there are no scoring plugins defined
### What happened?

Even when no scoring plugins are defined, scheduler always attempts to find multiple nodes where the pod can run. This is unnecessary (and therefore inefficient) when we know ahead of time that there will be no difference between how those nodes are scored. 

### What did you expect to happen?

If we're going to pick a random node, we can just as well cut short the search after finding the first match.

### How can we reproduce it (as minimally and precisely as possible)?

Configure scheduler without any scoring plugins and see that mulitple nodes are evaluated anyway.

### Anything else we need to know?

/cc @ahg-g @alculquicondor 
/sig scheduling
/kind bug

### Kubernetes version

1.29 and older

### Cloud provider

N/A

### OS version

_No response_

### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

_No response_

### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Skip looking for extra nodes if no scoring plugins are defined. As there's no difference in priority between nodes when we don't score them, there's no point in searching for multiple fitting nodes.

#### Which issue(s) this PR fixes:

Fixes #122057

#### Does this PR introduce a user-facing change?

```release-note
Improves scheduler performance when no scoring plugins are defined.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig scheduling
/cc @ahg-g

### CODE REVIEW

File: pkg/scheduler/schedule_one.go [hunk 1]
```diff
@@ -537,6 +537,9 @@ func (sched *Scheduler) findNodesThatPassFilters(
 	nodes []*framework.NodeInfo) ([]*v1.Node, error) {
 	numAllNodes := len(nodes)
 	numNodesToFind := sched.numFeasibleNodesToFind(fwk.PercentageOfNodesToScore(), int32(numAllNodes))
+	if !fwk.HasScorePlugins() {
```
Comment (alculquicondor) [inline pkg/scheduler/schedule_one.go:540]: can you also check that there aren't any score extenders?

Comment (aleksandra-malinowska) [reply pkg/scheduler/schedule_one.go]: I added checking if extender has `prioritizeVerb` configured - PTAL and let me know if that's alright.

**Comment by aleksandra-malinowska** [comment]
Failed test is a known flake: #121926

/retest

File: pkg/scheduler/extender_test.go [hunk 2]
```diff
@@ -93,6 +93,7 @@ func TestSchedulerWithExtenders(t *testing.T) {
 			registerPlugins: []tf.RegisterPluginFunc{
 				tf.RegisterFilterPlugin("TrueFilter", tf.NewTrueFilterPlugin),
 				tf.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				tf.RegisterScorePlugin("EqualPrioritizerPlugin", tf.NewEqualPrioritizerPlugin(), 20),
```
Comment (alculquicondor) [inline pkg/scheduler/extender_test.go:96]: can we have a unit test without a prioritizer where we see that `EvaluatedNodes` is one?

Comment (aleksandra-malinowska) [reply pkg/scheduler/extender_test.go:96]: added a new testcase, PTAL

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**